apps/src/codegen.js

```diff
@@ -468,12 +468,6 @@ exports.isNextStepSafeWhileUnwinding = function (interpreter) {
   if (state.done) {
     return true;
   }
-  if (type === "ForStatement") {
-    var mode = state.mode || 0;
-    // Safe to skip over ForStatement's in mode 0 (init) and 3 (update),
-    // but not mode 1 (test) or mode 2 (body) while unwinding...
-    return mode === 0 || mode === 3;
-  }
```

Comment:
We do now stop twice on the ForExpression while iterating the loop.  But the UI also now highlights the current mode (first `update` then `test`).